### ISSUE TITLE
Should native input validations avoided and be handled individually? 

### ISSUE THREAD

**Issue by ambrinchaudhary** [liferay/clay#4926]
## What is your question?
HTML input offers native validations such as **required**, **min**, **max**, **type=“number”**… that when the user submits the form, this is invalidated and it is not sent to backend, showing native error messages in the wrong inputs. 

If I am not wrong, these validations were handled in portal by taglibs such as `<aui:input>` where attributes like required, min, max … could be set,  along with `<aui:validator>` taglib where you could define common validations like for emails, digits, and so on. 

Now that we should render react components and do not use that taglibs, I don’t see we have anything similar in portal and we should implement these validations in each input or form we use (because we do not want the native messages to be shown). When having a form, the guide is to use Formik (https://github.com/liferay/liferay-frontend-projects/issues/348), but still the validations must be implemented in each component.

Also for the look and feel, we are setting the classes and managing the errors in each component we implement, like: 
```
<form>
      <div class={error ? ‘has-error’: ‘’}
	      <ClayInput {…props} >
      </div>

       {error && (
              <ClayForm.FeedbackGroup>
                  <ClayForm.FeedbackItem>
                      <ClayForm.FeedbackIndicator symbol="exclamation-full" />

                      {error}
                  </ClayForm.FeedbackItem>
              </ClayForm.FeedbackGroup>
          )}
</form>
```

I think this at least should be provided by Clay, avoiding us to implement this everywhere an input needs a validation. 

So my question here is: is this what we really want, to implement every validation whenever we need? Shouldn’t Clay or Portal should provide something for all this like we had previously in portal? Am I doing something wrong and all this should be handled in a different way that I am not aware of?

You can also find some context in this pull, https://github.com/liferay-lima/liferay-portal/pull/3116#issuecomment-1153566605 where I wanted to use the native **min** attribute for the size input but in order to avoid the native messages, I should implement this by myself. 
<img width="773" alt="Screenshot 2022-06-13 at 12 15 55" src="https://user-images.githubusercontent.com/8373764/173332532-f9286808-8408-4469-854e-93b8e2f4a492.png">

<img width="796" alt="Screenshot 2022-06-13 at 12 16 02" src="https://user-images.githubusercontent.com/8373764/173332542-dc0a3b37-9c9b-4ef4-a6d1-b9d4159012a8.png">

@lexicon

**Comment by matuzalemsteles** [comment]
Hey @ambrinchaudhary that's a good question, I'm going to answer some of your questions and an insight I have on it.

You can still get the native validations regardless if it's being written in React or not, https://codesandbox.io/s/quick-start-clay-finale-forked-bgjyku?file=/src/index.js. I believe that native and custom validations like `required`, `min`, and `max` would not need a custom error this would be a redundancy of error on the screen for the user in my opinion.

In my view the use of custom error is very difficult to offer because the validation depends on each implementation, the use of Formik is a recommendation only because it has very complete validation systems and manages forms with multiple fields very well, as well as managing the error status. That's one of the reasons why we don't implement everything in Clay, we don't want to build a new validation system from scratch or implement everything, we just want to allow you to use it with other libraries that the community uses and is much more developed and popular..

Another point about this might be the DX of the components, the way you declare errors for input, for example having to write a lot of composition to declare an error:

```jsx
<ClayForm.FeedbackGroup>
    <ClayForm.FeedbackItem>
        <ClayForm.FeedbackIndicator symbol="exclamation-full" />

        {error}
    </ClayForm.FeedbackItem>
</ClayForm.FeedbackGroup>
```

These are things that we want to simplify, we have an issue for this #4741 where we want to simplify the components related to the form, simplify both DX and offer OOTB features, such as more common validations, such as email and any other type of validation (required... with custom error), we still want to study how to offer that's because there are thousands of ways to implement this and thousands of libraries that implement it.

In summary, we want to improve this at least for the simplest cases but we will not replace libs like Formik for example, which manages the state of multiple fields.

For example, this would be a pseudocode example, but we still want to allow this to work when you declare it inside a Formik to do more complex validations or control multiple fields.

```jsx
<TextField
  validationState="valid"
  errorMessage=""
  label="Text"
/>

// or

<Field>
  <TextField label="Text"/>

  <Error validationState="valid" errorMessage="" />
</Field>
```

**Comment by matuzalemsteles** [comment]
@ambrinchaudhary thanks for your feedback. I will definitely be considering these use cases when we start working on this, I will also be notifying you when we have a draft proposal to gather feedback.

> The thing is that we don't always render the form via react, like my current case, where I only have some inputs and they are included in previous JSPs.

Oh I see, so something worked via JSP would also be desired! I'll think of something for that too.

**Comment by matuzalemsteles** [comment]
LPS Ticket https://liferay.atlassian.net/browse/LPS-190533
